roles/kubernetes/node/defaults/main.yml

```diff
@@ -92,6 +92,7 @@ loadbalancer_apiserver_memory_requests: 32M
 loadbalancer_apiserver_cpu_requests: 25m
 
 loadbalancer_apiserver_keepalive_timeout: 5m
+loadbalancer_apiserver_pod_name: NULL
```

Comment:
IMO you should either introduce two new variables for haproxy and nginx that defaults to their current name and directly use thosed (rather than checking is defined beforehand etc) or have a generic variable between the two like you do today with a jinja template similar to this (no tested/typo free): `{% if loadbalancer_apiserver_type == "nginx" %}nginx{% else %}haproxy`.

Reply:
I was hopeful to keep it to a single variable so folks could maintain the simplicity of a single target.  In theory I've got this sorted now...